codebase/config/sync/simplesamlphp_auth.settings.yml

```diff
@@ -8,14 +8,16 @@ login_link_show: true
 login_link_display_name: 'Federated login'
 header_no_cache: false
 role:
-  population: 'staff:eduPersonAffiliation,~=,STAFF'
+  population: 'global_admin:eduPersonPrincipalName,=,staff1@johnshopkins.edu|collection_level_admin:eduPersonPrincipalName,=,staff2@johnshopkins.edu'
```

Comment:
What worries me is this configuration will be backed into the image that will be deployed in the cloud. In that context: Would this population setting ever be triggered/used in production? i.e. do we depend on certain users (or certain classes of users, e.g. staff) to be assigned roles based on this setting?
* If so, would admins for cloud instances (e.g. prod) be expected to set this config value?
* If not, then is this 'staff1 = global admin" setting innocuous, as it'll never be triggered outside of the dev environment?

Reply:
Indeed this would be baked into the cloud image.

Since the principal names are unlikely to be valid in production, the setting is innocuous (no one would ever be able to login to Islandora as 'staff1' or 'staff2').

In production, these values will need to be set to values that are congruent with that environment.

Reply:
OK, if this needs to be set to something different in prod (or, more generally, different values for different environments,), then we'd probably want an environment variable for it.

Reply:
Probably a new issue to be dealt with separately?  Env vars working with external config is a new thing.

Reply:
Yeah.  I don't have a good understanding of what _should_ be in place for the system in production.  We haven't done env vars overriding config before, but it's something that is/will be part of the S3 PR (as it needs to do that).

An example config override in `local.settings.php`:
```
$config['s3fs.settings']['bucket'] = getenv('DRUPAL_DEFAULT_S3_BUCKET') ?: 'idc';
```
This overrides the 'bucket` config value for s3fs.

Reply:
A few weeks ago I noted some of this here https://github.com/jhu-idc/iDC-general/issues/62, but that ticket doesn't cover the _technical_ way we'd need to work with this.  

Sounds like it should be a separate issue because it does require some sort of technical fix. 

Should I make a ticket describing all of this?

Reply:
Yeah, a ticket for figuring out how to override the relevant config (and documenting the environment variables, or whatever is necessary to do so)

Reply:
https://github.com/jhu-idc/iDC-general/issues/318